src/crowbar.mli

```diff
@@ -1,52 +1,118 @@
 
 type ('k, 'res) gens =
   | [] : ('res, 'res) gens
   | (::) : 'a gen * ('k, 'res) gens -> ('a -> 'k, 'res) gens
+(** multiple generators are provided to functions using a listlike syntax.
+    for example, [map [int; int] (fun a b -> a + b)] *)
+
+type 'a printer = Format.formatter -> 'a -> unit
+(** pretty-printers for items generated by Crowbar; useful for the user in
+    translating test failures into bugfixes. *)
+
+(**/**)
 (* re-export stdlib's list
    We only want to override [] syntax in the argument to Map *)
 type nonrec +'a list = 'a list = [] | (::) of 'a * 'a list
+(**/**)
 
-val map : ('f, 'a) gens -> 'f -> 'a gen
+(** {1:top Generators } *)
+
+(** {2 Simple Generators } *)
 
+val int : int gen
+(** [int] generates an integer.  For 32-bit systems, it will generate an
+    Int32.t; otherwise, it will generate an Int64.t .  If you need integers from
+    a smaller domain, consider using {!range}. *)
+val uint8 : int gen
+(** [uint8] generates an unsigned byte, ranging from 0 to 255 inclusive. *)
+val int8 : int gen
+(** [int8] generates a signed byte, ranging from -128 to 127 inclusive. *)
+val int32 : Int32.t gen
+(** [int32] generates a 32-bit signed integer. *)
+val int64 : Int64.t gen
+(** [int64] generates a 64-bit signed integer. *)
+val float : float gen
+(** [float] generates a double-precision floating-point number. *)
+val bytes : string gen
+(** [bytes] generates a string of arbitrary length (including zero-length strings).  *)
+val bool : bool gen
+(** [bool] generates a yes or no answer. *)
+val range : int -> int gen
+(** [range n] is a generator for integers between 0 (inclusive) and [n] (exclusive). *)
+
+(** {2 Functions on Generators } *)
+
+val map : ('f, 'a) gens -> 'f -> 'a gen
+(** [map gens map_fn] provides a means for creating generators using other
+    generators' output.  For example, one might generate a Char.t from a
+    {!uint8}:
+    {[
+      open Crowbar
+      let char_gen : Char.t gen = map [uint8] Char.chr
+    ]}
+    *)
 val join : 'a gen gen -> 'a gen
 
 val bind : 'a gen -> ('a -> 'b gen) -> 'b gen
 
 val unlazy : 'a gen Lazy.t -> 'a gen
+(** [unlazy gen] forces the generator [gen].  It is useful when defining
+    generators for recursive data types:
+
+    {[
+      open Crowbar
+      type a = A of int | Self of a
+      let rec a_gen = lazy (
+        choose [
+          map [int] (fun i -> A i);
+          map [(unlazy a_gen)] (fun s -> Self s);
+        ])
+      let lazy a_gen = a_gen
+    ]}
+    *)
 val fix : ('a gen -> 'a gen) -> 'a gen
+(** [fix fn] applies the function [fn].  It is useful when defining generators
+    for recursive data types:
+
+    {[
+      open Crowbar
+      type a = A of int | Self of a
+      let rec a_gen = fix (fun a_gen ->
+          choose [
+          map [int] (fun i -> A i);
+          map [a_gen] (fun s -> Self s);
+        ])
+    ]}
+    *)
 
 val const : 'a -> 'a gen
+(** [const a] creates a generator which always generates [a]. *)
```

Comment:
I think you should stick to doc string pattern to describe `'a gen` values and stick to it. Either:

1. "`bla` is a generator that/for" or 
2. "`bla` generates..." (as above)